src/group_theory/order_of_element.lean

```diff
@@ -684,3 +688,28 @@ subgroup_of_idempotent (S ^ (fintype.card G)) ⟨1, one_mem⟩ begin
 end
 
 end pow_is_subgroup
+
+section linear_ordered_ring
+
+variable [linear_ordered_ring G]
+
+lemma order_of_abs_ne_one (h : |x| ≠ 1) : order_of x = 0 :=
+begin
+  rw order_of_eq_zero_iff',
+  intros n hn hx,
+  replace hx : |x| ^ n = 1 := by simpa only [abs_one, abs_pow] using congr_arg abs hx,
+  cases h.lt_or_lt with h h,
+  { exact ((pow_lt_one (abs_nonneg x) h hn.ne').ne hx).elim },
+  { exact ((one_lt_pow h hn.ne').ne' hx).elim }
+end
+
+lemma linear_ordered_ring.order_of_le_two : order_of x ≤ 2 :=
+begin
+  cases ne_or_eq (|x|) 1 with h h,
+  { convert zero_le_two, exact order_of_abs_ne_one h },
```

Comment:
```suggestion
  { convert zero_le_two,
    exact order_of_abs_ne_one h },
```
Each tactic application goes on its own line.

Reply:
Just for style, some other suggestions to choose from (untested):
```suggestion
  { simpa [order_of_abs_ne_one h] using zero_le_two }
```
(a one-liner);
```suggestion
  { rw order_of_abs_ne_one h,
    norm_num }
```
(puts more focus on the lemma with content by automating away the lemma without content)

Reply:
turns out `simp` does it on its own! thanks :)